Sources/SourceKitLSP/DocumentManager.swift

```diff
@@ -153,18 +153,16 @@ public final class DocumentManager {
           // Remove all tokens in the updated range and shift later ones.
           let rangeAdjuster = RangeAdjuster(edit: edit)!
 
-          document.latestTokens.withMutableTokensOfEachKind { tokens in
-            tokens = Array(tokens.lazy
-              .compactMap {
-                var token = $0
-                if let adjustedRange = rangeAdjuster.adjust(token.range) {
-                  token.range = adjustedRange
-                  return token
-                } else {
-                  return nil
-                }
-              })
-          }
+          document.latestTokens.semantic = Array(document.latestTokens.semantic.lazy
+            .compactMap {
```

Comment:
It will not have a positive effect since we're immediately draining the results into a fresh array to store it here. I left this code alone when I migrated the lexical token stream, so I'll drop it in #651.